week-8/Homework/mandatory/4-programmer-humour/script.js

```diff
@@ -0,0 +1,20 @@
+const api_url = "https://xkcd.now.sh/?comic=latest";
+
+let jokeImage = document.getElementById("joke");
+let jokeTitle = document.getElementById("jokeTitle");
+
+function getJoke() {
+  fetch(api_url)
+    .then((response) => response.json())
+    .then((joke) => {
+      console.log(joke);
+      jokeTitle.innerText = joke.title;
+      jokeImage.src = joke.img;
+      jokeImage.alt = joke.alt;
+    })
+    .catch((error) => {
+      console.log("error!!!");
+      console.error(error);
+    });
+}
+getJoke();
```

Comment:
This function is correct, calling the function at line 20 is not appropriate. what you could do separate the fetching and the getJoke then call getJoke in the .then() or the easiest way apply the innerText in the .then() . Work on that.